### PR TITLE
feat: [Redis] expose automatic failover enabled

### PR DESCRIPTION
**Pull request by fnaranjo-vmw**
[#184471687](https://www.pivotaltracker.com/story/show/184471687)

This PR depends on: https://github.com/cloudfoundry/cloud-service-broker/pull/779

Removed one integration test since it no longer applies. automatic_failover_enabled is no longer computed based on node_count value

### Checklist:

* [x] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: aws-redis.yml [hunk 1]
```diff
@@ -113,6 +113,12 @@ provision:
     details: The ARN of the key to use if encrypting at rest. Defaults to AWS managed key.
     default: ""
     prohibit_update: true
+  - field_name: automatic_failover_enabled
+    type: boolean
+    details: Automatically promote replica to primary if the existing primary fails. If enabled, node_count must be greater than 1.
+    default: true
+    required: false
+    prohibit_update: false
```
[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply aws-redis.yml]: Fixed in: https://github.com/cloudfoundry/csb-brokerpak-aws/pull/797/commits/7471616ff5d897b5bbfd62ddca5dc4d4e6a4eb89

File: aws-redis.yml [hunk 2]
```diff
@@ -113,6 +113,12 @@ provision:
     details: The ARN of the key to use if encrypting at rest. Defaults to AWS managed key.
     default: ""
     prohibit_update: true
+  - field_name: automatic_failover_enabled
+    type: boolean
+    details: Automatically promote replica to primary if the existing primary fails. If enabled, node_count must be greater than 1.
+    default: true
+    required: false
```
[the remark this answers is not in the document]
Comment (fnaranjo-vmw) [reply aws-redis.yml]: Fixed in: https://github.com/cloudfoundry/csb-brokerpak-aws/pull/797/commits/7471616ff5d897b5bbfd62ddca5dc4d4e6a4eb89

**Review by zucchinidev — APPROVED** [review]

File: aws-redis.yml [hunk 3]
```diff
@@ -226,6 +230,12 @@ examples:
   plan_id: c7f64994-a1d9-4e1f-9491-9d8e56bbf146
   provision_params: {"region": "us-east-1"}
   bind_params: {}
+- name: invalid-automatic-failover
+  description: Try to set `automatic_failover_enabled` with `node_count` < 2
+  expected_error: When using automatic failover, there must be at least 2 cache clusters in the replication group
+  plan_id: c7f64994-a1d9-4e1f-9491-9d8e56bbf146
+  provision_params: {"node_count": 1, "automatic_failover_enabled": true}
+  bind_params: {}
```
Comment (fnaranjo-vmw) [inline aws-redis.yml:238]: By simply removing these 6 lines, this PR will no longer depend on https://github.com/cloudfoundry/cloud-service-broker/pull/779

Consider doing that if there is an urgency to release this change, and there is reluctancy in merging the feature to support failing examples.

Comment (fnaranjo-vmw) [reply aws-redis.yml:238]:
```suggestion
```

**Pull request merged**